src/os/bluestore/BlueStore.cc

```diff
@@ -14784,8 +14925,44 @@ void BlueStore::BlueStoreThrottle::finish_start_transaction(
 {
   ceph_assert(txc.deferred_txn);
   throttle_deferred_bytes.get(txc.cost);
+  emit_initial_tracepoint(db, txc, start_throttle_acquire);
 }
 
+#if defined(WITH_LTTNG)
+void BlueStore::BlueStoreThrottle::complete_kv(TransContext &txc)
+{
+  pending_kv_ios -= 1;
+  ios_completed_since_last_traced++;
+  if (txc.tracing) {
+    tracepoint(
+      bluestore,
+      transaction_commit_latency,
+      txc.osr->get_sequencer_id(),
+      txc.seq,
+      ((double)ceph_clock_now()) - ((double)txc.start));
+  }
+}
+#endif
+
+#if defined(WITH_LTTNG)
+void BlueStore::BlueStoreThrottle::complete(TransContext &txc)
+{
+  if (txc.deferred_txn) {
+    pending_deferred_ios -= 1;
+  }
+  if (txc.tracing) {
+    utime_t now = ceph_clock_now();
+    double usecs = (now.to_nsec()-txc.start.to_nsec())/1000;
```

Comment:
shall we cast `now.to_nsec()-txc.start.to_nsec()` to double before dividing it by 1000 ?

also, might want to add spaces around the operators.